src/components/comment-input.tsx

```diff
@@ -20,11 +22,31 @@ const InputContainer = styled.div`
 `;
 
 export const CommentInput: React.VoidFunctionComponent = () => {
+  const { setNewCommentText, newCommentText, comments, setComments } =
+    useComments();
+
+  const onSend = () => {
+    const comment: Comment = {
+      id: comments.length,
+      user: "anon",
+      text: newCommentText,
+      date: new Date().toUTCString(),
+    };
+    setComments([...comments, comment]);
+    setNewCommentText("");
+  };
+
   return (
     <InputContainer>
       <AvatarImage />
-      <TextInput placeholder="Добавить комментарий..." />
-      <Button>Отправить</Button>
+      <TextInput
+        placeholder="Добавить комментарий..."
+        value={newCommentText}
+        onChange={(e: ChangeEvent<HTMLInputElement>) =>
```

Comment:
TypeScript can derive the type of `e` because `onChange` is typed and the function is inline. So strictly speaking, we don’t need to specify the type of `e`.

<img width="712" alt="Screenshot 2021-12-14 at 09 04 50" src="https://user-images.githubusercontent.com/608862/145967442-5e7b375d-5b69-43af-9163-f008df073e49.png">

I’ll see if I can add a lint rule for that some day.

We can also lint variable name length because `e` can refer to `error`, `event`, etc. By being more specific (`event`) we help ourselves read someone else’s code, especially long chunks of logic.

Reply:
Sure, will rename it. As a general rule, the variable length depends on the scope - as smaller the scope, as shorter can be the name, as it's closer to the point of implementation.